pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -0,0 +1,464 @@
+		return reconcile.Result{}, err
+	}
+	if authenticationCR != nil && authenticationCR.Status.State != operatorv1.TigeraStatusReady {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Authentication is not ready - authenticationCR status: %s", authenticationCR.Status.State), nil, reqLogger)
+		return reconcile.Result{}, nil
+	}
+
+	// Create a component handler to manage the rendered component.
+	handler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
+
+	keyValidatorConfig, err := utils.GetKeyValidatorConfig(ctx, r.client, authenticationCR, r.clusterDomain)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "Failed to process the authentication CR.", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	reqLogger.V(3).Info("rendering components")
+	hasNoLicense := !utils.IsFeatureActive(license, common.PolicyRecommendationFeature)
+	openshift := r.provider == operatorv1.ProviderOpenShift
+	policyRecommendationCfg := &render.PolicyRecommendationConfiguration{
+		ClusterDomain:                        r.clusterDomain,
+		ESClusterConfig:                      esClusterConfig,
+		ESSecrets:                            esSecrets,
+		HasNoLicense:                         hasNoLicense,
+		Installation:                         network,
+		KeyValidatorConfig:                   keyValidatorConfig,
+		ManagementClusterConnection:          managementClusterConnection,
+		PolicyRecommendationServerCertSecret: policyRecommendationServerCertSecret,
+		PullSecrets:                          pullSecrets,
+		TrustedBundle:                        trustedBundle,
+		Openshift:                            openshift,
+		UsePSP:                               r.usePSP,
+	}
+
+	// Render the desired objects from the CRD and create or update them.
+	component := render.PolicyRecommendation(policyRecommendationCfg)
+
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, component); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	components := []render.Component{
+		component,
+		rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
+			Namespace:       render.PolicyRecommendationNamespace,
+			ServiceAccounts: []string{render.PolicyRecommendationName},
+			KeyPairOptions: []rcertificatemanagement.KeyPairOption{
+				rcertificatemanagement.NewKeyPairOption(policyRecommendationCfg.PolicyRecommendationServerCertSecret, true, true),
+			},
+			TrustedBundle: trustedBundle,
+		}),
+	}
+
+	for _, cmp := range components {
+		if err := handler.CreateOrUpdateOrDelete(context.Background(), cmp, r.status); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	if hasNoLicense {
```

Comment:
I think this should come before rendering the components on line 435, otherwise it is too late.